### ISSUE TITLE
I can't get toEventually to work correctly after updating to Nimble 11.x

### ISSUE THREAD

**Issue by marcinpodeszwa-medcase** [Quick/Nimble#1015]
- [X] I have read [CONTRIBUTING](https://github.com/Quick/Nimble/blob/main/CONTRIBUTING.md) and have done my best to follow them.

### What did you do?

Consider following sample code:
```
class Sample {
    static var executed = false

    func execute() {
        DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
            Self.executed = true
        }
    }
}
```

I want to test it using Quick+Nimble, so I wrote following test:
```
class SampleTests: QuickSpec {
    override func spec() {
        describe("Sample") {
            it("sets executed") {
                Sample().execute()
                expect(Sample.executed).toEventually(beTrue())
            }
        }
    }
}
```

And it was working perfectly well on **Quick 5.0.1** and **Nimble 10.0.0**

After updating Quick to 6.0.0 and Nimble to 11.1.0 I saw this warning (but the test was passing):
<img width="657" alt="Screenshot 2022-11-07 at 15 24 07" src="https://user-images.githubusercontent.com/95348173/200334880-d20c5586-7de6-46a5-a152-fadbba1ccf68.png">

I added keyword `await` (that's what I thought I need to do) and suddenly my test stopped passing:
<img width="715" alt="Screenshot 2022-11-07 at 15 24 25" src="https://user-images.githubusercontent.com/95348173/200335194-eaa83295-dd12-44fb-bce3-d581c311e919.png">

### What did you expect to happen?

I expect this test to pass without warnings. Maybe there is something I don't understand, because I'm new to new swift's concurrency.

### What actually happened instead?

Depending on if I add `await` or not, I get either a warning or a test failure.

### Environment

List the software versions you're using:

- Quick: *6.0.0*
- Nimble: *11.1.0*
- Xcode Version: *Version 14.1 (14B47b)* (Open Xcode; In menubar: Xcode > About Xcode)
- Swift Version: *5.7.1* (Open Xcode Preferences; Components > Toolchains. If none, use `Xcode Default`.)

Please also mention which package manager you used and its version. Delete the
other package managers in this list:

- Swift Package Manager *5.7.1* (Use `swift build --version` in Terminal)

### Project that demonstrates the issue

Please link to a project we can download that reproduces the issue. Feel free
to delete this section if it's not relevant to the issue (eg - feature request).

[ToEventuallyTest.zip](https://github.com/Quick/Nimble/files/9952617/ToEventuallyTest.zip)

The project should be [short, self-contained, and correct example](http://sscce.org/).

**Comment by younata** [comment]
Hey, thanks for the succinct bug report with reproducible sample!

This is odd to me. The `Self.executed = true` line is getting executed. The static var is getting set. But it appears that Nimble is sending a cached value to the matcher? Wild.

**Comment by younata** [comment]
This fix is included in the [just-released v11.1.1](https://github.com/Quick/Nimble/releases/tag/v11.1.1)!

**Comment by marcinpodeszwa-medcase** [comment]
thanks @younata , works great now!
